runner/src/server/plugins/engine/components/ClientSideFileUploadField.ts

```diff
@@ -0,0 +1,83 @@
+import { FormComponent } from "./FormComponent";
+import { FormData, FormSubmissionErrors } from "../types";
+import { ClientSideFileUploadFieldComponent } from "@xgovformbuilder/model";
+import { FormModel } from "server/plugins/engine/models";
+import { ClientSideFileUploadFieldViewModel } from "server/plugins/engine/components/types";
+import joi from "joi";
+export class ClientSideFileUploadField extends FormComponent {
+  options: ClientSideFileUploadFieldComponent["options"];
+  schema: ClientSideFileUploadFieldComponent["schema"];
+
+  constructor(def: ClientSideFileUploadFieldComponent, model: FormModel) {
+    super(def, model);
+    this.options = def.options;
+    this.schema = def.schema;
+  }
+
+  getFormSchemaKeys() {
+    return {
+      [this.name]: joi.allow("").allow(null),
+      [`${this.name}__filename`]: joi.string().optional(),
+      [`${this.name}__delete[]`]: joi.string().optional(),
```

Comment:
All fields are optional, we have custom validation below that ensures the state of the bucket reflects our necessary options.

the `__filename` & `__delete[]` are for no JavaScript fallback users, the way the component is written, you can still select and delete files without JavaScript, you just won't be able to see the progress of them due to not having JavaScript.